Add new_test/5.1/teams/test_target_teams_thread_limit.F90

seyonglee opened the pull request:
[Results on Summit]
- GCC 13.1.1: 
    - Both C and Fortran tests passed.
- XL 16.1.1-10:
    - C test passed but ran on the host with the following warning: The number of teams was unexpected, the test results are likely inconclusive
    - Fortran test failed: line 45.48: 1515-022 (S) Syntax Error: Extra token " thread_limit " was found. The token is ignored.
- NVHPC 22.11:
    - C test failed: line 39: error: invalid text in pragma (in thread_limit)
    - Fortran test failed: NVFORTRAN-S-0533-Clause 'THREAD_LIMIT' not allowed in OMP TARGET
- LLVM 17.0.0: 
    - C test passed.

spophale commented:
@seyonglee since both target and teams accepts thread_limit, we should rename this test to test_target_thread_limit[_teams] and have a separate test for teams +thread_limt. Thoughts ?

seyonglee commented:
> @seyonglee since both target and teams accepts thread_limit, we should rename this test to test_target_thread_limit[_teams] and have a separate test for teams +thread_limt. Also, not sure which folder this goes in.. target or teams.
> 
> Thoughts ?

How about this?
1) Copy these tests, rename the copies as `test_target_thread_limit`, and move them to` test/5.1/target` directory.
2) In these original tests, move `thread_limit` clause to the teams directive.

fel-cab commented:
> > @seyonglee since both target and teams accepts thread_limit, we should rename this test to test_target_thread_limit[_teams] and have a separate test for teams +thread_limt. Also, not sure which folder this goes in.. target or teams.
> > Thoughts ?
> 
> How about this?
> 
> 1. Copy these tests, rename the copies as `test_target_thread_limit`, and move them to` test/5.1/target` directory.
> 2. In these original tests, move `thread_limit` clause to the teams directive.

I think it is a good solution

seyonglee commented:
Updated the following things:
        - Move the thread_limit clause from the target directive to the teams directive.
        - Changed the comments accordingly.
        - If the num_teams is bigger than OMPVV_NUM_TEAMS_DEVICE, the array out of bounds error can occur; fixed this issue.

seyonglee commented:
[New results on Summit]
- GCC 13.1.1: 
    - Both C and Fortran tests passed.
- XL 16.1.1-10:
    - C test passed but ran on the host with the following warning: The number of teams was unexpected, the test results are likely inconclusive
    - Fortran test passed.
- NVHPC 22.11:
    - C test failed: The number of teams was unexpected, the test results are likely inconclusive; The value of sum_errors is 8.
    - Fortran test passed.
- LLVM 17.0.0: 
    - C test passed.
